cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/BusinessMetadataStore.java

```diff
@@ -48,11 +56,17 @@
 
   private static final Id.DatasetInstance BUSINESS_METADATA_INSTANCE_ID =
     Id.DatasetInstance.from(Id.Namespace.SYSTEM, "business.metadata");
+  private static final Map<String, String> EMPTY_PROPERTIES = ImmutableMap.of();
+  private static final Set<String> EMPTY_TAGS = ImmutableSet.of();
+
   private final Transactional<BusinessMdsIterable, BusinessMetadataDataset> txnl;
+  private final boolean publishEnabled;
+  private final MetadataChangePublisher changePublisher;
```

Comment:
There is an implicit assumption here that publishEnabled is false iff changePublisher is NoOpChangePublisher. If that does not hold, this class might crash all over the place.